snap/README.md

```diff
@@ -114,35 +162,335 @@ Or by using the systemd unit name and `journalctl`:
+
+```bash
+$ sudo snap set edgexfoundry env.security-proxy.tls-certificate=""
+$ sudo snap set edgexfoundry env.security-proxy.tls-private-key=""
+```
+
+If you are using a different server name than localhost, then it needs to be specified first using the optional tls-sni configuration setting. Example:
+
+```bash
+$ edgeca gencsr --cn server01 --csr csrfile --key csrkeyfile
+$ edgeca gencert -o server.cert -i csrfile -k server.key
+$ sudo snap set edgexfoundry env.security-proxy.tls-certificate=""
+$ sudo snap set edgexfoundry env.security-proxy.tls-private-key=""
+$ sudo snap set edgexfoundry env.security-proxy.tls-sni="server01"
+$ sudo snap set edgexfoundry env.security-proxy.tls-certificate="$(cat server.cert)"
+$ sudo snap set edgexfoundry env.security-proxy.tls-private-key="$(cat server.key)"
+
+$ curl -v --cacert /var/snap/edgeca/current/CA.pem -X GET https://server01:8443/coredata/api/v1/ping? -H "Authorization: Bearer $TOKEN"
 ```
-**Note** - Kong is currently not supported in the snap when installed on an arm64-based device, so it will be disabled on install.
 
 ## Limitations
 
 [See the GitHub issues with label snap for current issues.](https://github.com/edgexfoundry/edgex-go/issues?q=is%3Aopen+is%3Aissue+label%3Asnap)
 
+## Service environment configuration overrides
+**Note** - all of the configuration options below must be specified with the prefix: 'env.<service>.' where '<service>' is one of the following:
+
+  - core-command
+  - core-data
+  - core-metadata
+  - support-notifications
+  - support-scheduler
+  - sys-mgmt-agent
+  - security-proxy
+  - security-secret-store
+  - device-virtual
+  - app-service-config
+
+```
+[Service]
+service.boot-timeout            // Service.BootTimeout
+service.check-interval          // Service.CheckInterval
+service.host                    // Service.Host
+service.server-bind-addr        // Service.ServerBindAddr
+service.port                    // Service.Port
+service.protocol                // Service.Protocol
+service.max-result-count        // Service.MaxResultCount
+service.read-max-limit          // Service_ReadMaxLimit         // app-service-configurable only
+service.startup-msg             // Service.StartupMsg
+service.timeout                 // Service.Timeout
+
+[Clients.CoreData]
+clients.data.port                 // Clients.CoreData.Port
+
+[Clients.MetaData]
+clients.metadata.port             // Clients.MedataData.Port
+
+[Clients.Notifications]
+clients.notifications.port        // Clients.Notifications.Port
+
+[Clients.Scheduler]
+clients.scheduler.port            // Clients.Scheduler.Port   // sys-mgmt-only
+
+[MessageQueue]
+messagequeue.topic                // MessageQueue.Topic       // core-data only
+
+[SecretStore]
+secretstore.additional-retry-attempts    // SecretStore.AdditionalRetryAttempts
+secretstore.retry-wait-period            // SecretStore.RetryWaitPeriod
+```
+
+### API Gateway settings (prefix: env.security-proxy.)
```

Comment:
again should this include `-setup`?

Reply:
yes, updated this as a part of an overall update of this section